backend/Recodo.API/Middleware/ErrorHandlerMiddleware.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Microsoft.AspNetCore.Http;
+using Recodo.BLL.Exceptions;
+using Recodo.Common.Error;
+using System;
+using System.Net;
+using System.Threading.Tasks;
+
+namespace Recodo.API.Middleware
+{
+    public class ErrorHandlerMiddleware
+    {
+        private readonly RequestDelegate _next;
+
+        public ErrorHandlerMiddleware(RequestDelegate next)
+        {
+            _next = next;
+        }
+
+        public async Task Invoke(HttpContext context)
+        {
+            try
+            {
+                await _next(context);
+            }
+            catch (Exception error)
+            {
+                var response = context.Response;
+                response.ContentType = "application/json";
+
+                switch (error)
+                {
+                    case NotFoundException:
+                        response.StatusCode = (int)HttpStatusCode.NotFound;
+                        break;
+                    case InvalidUserNameOrPasswordException:
+                        response.StatusCode = (int)HttpStatusCode.Unauthorized;
+                        break;
+                    case ExpiredRefreshTokenException:
+                        response.StatusCode = (int)HttpStatusCode.Unauthorized;
+                        break;
+                    case InvalidTokenException:
+                        response.StatusCode = (int)HttpStatusCode.Unauthorized;
+                        break;
+                    default:
+                        response.StatusCode = (int)HttpStatusCode.InternalServerError;
+                        break;
+                }
```

Comment:
```suggestion
                response.StatusCode = error switch
                {
                    NotFoundException => (int)HttpStatusCode.NotFound,
                    InvalidUserNameOrPasswordException => (int)HttpStatusCode.Unauthorized,
                    ExpiredRefreshTokenException => (int)HttpStatusCode.Unauthorized,
                    InvalidTokenException => (int)HttpStatusCode.Unauthorized,
                    _ => (int)HttpStatusCode.InternalServerError
                };
```

Let's use modern language features 😉